pkg/process/checks/container_check_test.go

```diff
@@ -0,0 +1,148 @@
+
+	// Set small size per chunk to encourage chunking behavior
+	containerCheck.maxBatchSize = 1
+
+	ctr1 := &model.Container{
+		Id: "container-1",
+		Addresses: []*model.ContainerAddr{
+			{
+				Ip:       "10.0.2.15",
+				Port:     32769,
+				Protocol: model.ConnectionType_tcp,
+			},
+			{
+				Ip:       "172.17.0.4",
+				Port:     6379,
+				Protocol: model.ConnectionType_tcp,
+			},
+		},
+	}
+	ctr2 := &model.Container{
+		Id: "container-2",
+		Addresses: []*model.ContainerAddr{
+			{
+				Ip:       "172.17.0.2",
+				Port:     80,
+				Protocol: model.ConnectionType_tcp,
+			},
+		},
+	}
+
+	containers := []*model.Container{ctr1, ctr2}
+	mockContainerProvider.EXPECT().GetContainers(cacheValidityNoRTTest, nil).Return(containers, nil, nil, nil)
+
+	expected := []model.MessageBody{
+		&model.CollectorContainer{
+			Containers: []*model.Container{ctr1},
+			Info:       containerCheck.hostInfo.SystemInfo,
+			GroupSize:  int32(len(containers)),
+		},
+		&model.CollectorContainer{
+			Containers: []*model.Container{ctr2},
+			Info:       containerCheck.hostInfo.SystemInfo,
+			GroupSize:  int32(len(containers)),
+		},
+	}
+
+	// Test check runs without error
+	actual, err := containerCheck.Run(func() int32 { return 0 }, &RunOptions{RunStandard: true, NoChunking: false})
+	require.NoError(t, err)
+	assert.ElementsMatch(t, expected, actual.Payloads())
+}
+
+func TestContainerCheckWithoutChunking(t *testing.T) {
+	containerCheck, mockContainerProvider := containerCheckWithMockContainerProvider(t)
+
+	// Set small size per chunk to encourage chunking behavior
+	containerCheck.maxBatchSize = 1
+
+	ctr1 := &model.Container{
+		Id: "container-1",
+		Addresses: []*model.ContainerAddr{
+			{
+				Ip:       "10.0.2.15",
+				Port:     32769,
+				Protocol: model.ConnectionType_tcp,
+			},
+			{
+				Ip:       "172.17.0.4",
+				Port:     6379,
+				Protocol: model.ConnectionType_tcp,
+			},
+		},
+	}
+	ctr2 := &model.Container{
+		Id: "container-2",
+		Addresses: []*model.ContainerAddr{
+			{
+				Ip:       "172.17.0.2",
+				Port:     80,
+				Protocol: model.ConnectionType_tcp,
+			},
+		},
+	}
+
+	containers := []*model.Container{ctr1, ctr2}
+	mockContainerProvider.EXPECT().GetContainers(cacheValidityNoRTTest, nil).Return(containers, nil, nil, nil)
+
+	// Test check runs without error
+	actual, err := containerCheck.Run(func() int32 { return 0 }, &RunOptions{RunStandard: true, NoChunking: true})
+	require.NoError(t, err)
+
+	// Assert to check there is only one chunk and that the nested values of this chunk match expected
+	assert.Len(t, actual.Payloads(), 1)
+	actualPayloads := actual.Payloads()
+	assert.IsType(t, &model.CollectorContainer{}, actualPayloads[0])
+	collectorContainer := actualPayloads[0].(*model.CollectorContainer)
+	ProcessDiscoveries := collectorContainer.GetContainers()
+	assert.ElementsMatch(t, containers, ProcessDiscoveries)
+	assert.EqualValues(t, 1, collectorContainer.GetGroupSize())
```

Comment:
why not just use an `assert.elements.Match()` with an expected match similar to the previous tests? Also these two tests look very similar. You can use table tests to reduce the copy/paste.

Reply:
Example of a table tests: https://github.com/DataDog/datadog-agent/blob/c9412b7370e75e123880a98755843106605ee56b/pkg/config/setup/process_test.go#L578-L608

Reply:
I tried assert.ElementsMatch() but it only matches at the highest level and does not match nested or "deeper" fields ignoring their order (additional context if you are interested: https://github.com/stretchr/testify/issues/676). So since no chunking tests put all processes/containers into one chunk, it groups these objects into a nested slice field within the collector, which can't be matched. 

For integrating table tests, if the asserts are different, is it still recommended?

Reply:
So I think you just need to set this up so you have an `expectedContainers [][]*model.Container` and you assert that the message has as many payloads and the elements in the payloads match.

Reply:
Marking conversation as resolved - table tests were implemented.